packages/flutter_tools/lib/src/ios/xcresult.dart

```diff
@@ -361,7 +421,38 @@ bool _shouldDiscardIssue({
   return false;
 }
 
-List<XCResultIssue> _parseIssuesFromIssueSummariesJson({
+/// Helper to parse issues from the new (Xcode 16+) flat list format.
+List<XCResultIssue> _parseIssuesFromNewFormat({
+  required XCResultIssueType type,
+  required Object? jsonList,
+  required List<XCResultIssueDiscarder> issueDiscarders,
+}) {
+  final issues = <XCResultIssue>[];
+  if (jsonList is! List<Object?>) {
+    return issues;
+  }
+
+  for (final Object? issueJson in jsonList) {
+    if (issueJson is! Map<String, Object?>) {
+      continue;
+    }
+    final resultIssue = XCResultIssue.fromNewFormat(type: type, issueJson: issueJson);
+    var discard = false;
+    for (final discarder in issueDiscarders) {
+      if (_shouldDiscardIssue(issue: resultIssue, discarder: discarder)) {
+        discard = true;
+        break;
+      }
+    }
+    if (!discard) {
+      issues.add(resultIssue);
+    }
+  }
+  return issues;
```

Comment:
This is great! You don't have to do this,  but I prefer `where` and `map` when editing collections b/c I feel like it's easier to read. 

eg.
```
  return jsonList
      .whereType<Map<String, Object?>>() 
      .map((issueJson) => XCResultIssue.fromNewFormat( 
          type: type,
          issueJson: issueJson)
      .where((issue) { 
        final shouldDiscard = issueDiscarders.any(
          (discarder) => _shouldDiscardIssue(issue: issue, discarder: discarder),
        );
        return !shouldDiscard;
      })
)
```
Thats mostly pseudo code but something like that?